### ISSUE TITLE
Android app crashing

### ISSUE THREAD

**Issue by SteveScott** [zulip/zulip-mobile#5757]
For the past month, as I am typing on my Android Pixel 7 Pro, Zulip crashes. No message or notification. I re-load the app and my text is still there. What can I do to help troubleshoot or report this bug? Version 27.211.

**Comment by chrisbobbe** [comment]
Hi @SteveScott, thanks for the report! What version of Android are you on?

**Comment by SteveScott** [comment]
I am running Android 13

**Comment by chrisbobbe** [comment]
Hmm, I wonder if this is #5741.

**Comment by kylehousley** [comment]
I have also been experiencing the same issue for the past couple weeks, on Android 13 as well.

**Comment by chrisbobbe** [comment]
We're aware that Android crashes have been becoming much more frequent recently, especially on Google Pixel devices, and we're investigating the cause.

**Comment by maltokyo** [comment]
I also have this, started recently (in last two or three days) on Xiaomi phone running latest version of Android 13. Crashes now every time I open the app. How can I get logs to share, I can't work it out. 

Tried deleting cache and all app data and logging in again, same issue.

**Comment by lakinwecker** [comment]
I've been experiencing crashes for the past 2ish weeks as well, Android 13, almost always when typing, sometimes when not. Other lichess team members are also all starting to experience the same crashes as well.

**Comment by L-Marcel** [comment]
> I also have this, started recently (in last two or three days) on Xiaomi phone running latest version of Android 13. Crashes now every time I open the app. How can I get logs to share, I can't work it out. 
> 
> Tried deleting cache and all app data and logging in again, same issue. 

Me too.

**Comment by chrisbobbe** [comment]
Thank you all for your patience. Debugging continues in the [Zulip development community](https://zulip.com/development-community/), [here](https://chat.zulip.org/#narrow/stream/48-mobile/topic/Android.20crashes/near/1640893).

Quoting @gnprice from there ([link](https://chat.zulip.org/#narrow/stream/48-mobile/topic/Android.20crashes/near/1642685)):

> For users who are technically sophisticated and willing to help debug, I think probably the most useful thing is to use `adb logcat` to get a crash dump.
> 
> Note that the Android logs have a "crash" section which is separate from the main section. The stack trace is likely to be in the "crash" section, and is the most interesting piece to get.

> Those would help us see if we're seeing the same stack trace all the time, or several of them. And the stack traces might have something in them that helps identify a piece of the system that's likely to be involved.

> The other most likely fruitful path to pursue is probably to study the [[Google Play system update changes](https://chat.zulip.org/#narrow/stream/48-mobile/topic/Android.20crashes/near/1642674)] to see what happened between the July and August updates that might plausibly trigger something like this.
> 
> Then that information can feed into that [[thread](https://issuetracker.google.com/issues/300840851)] in the Android tracker, and may help get it more quickly routed to the right team, or more quickly diagnosed once it is, and then fixed.

> Knowing the system involved, or a diagnosis, might also give us an idea of something we could do to avoid triggering the issue.
> 
> Or it might give an idea of something RN could do to avoid triggering it.

We suspect that a Google Play system update from August 1 is a factor in this crash. If you tap Settings > About phone > Android version, you should see two dates, like this:

> Android security update: August 5, 2023
> Google Play system update: August 1, 2023

For anyone who's been seeing this issue's symptoms—crashes that started or became frequent recently—you may want to check that "Google Play system update". We expect you'll see August 1 there. We don't yet have reports from anyone on a September 1 update, and we'd like to know if the problem persists with that update.

**Comment by russp41** [comment]
I've been experiencing this issue since Sept 14. Zulip crashes frequently on Samsung Galaxy S23 Ultra, Android 13, "Google Play system update" date is August 1.

**Comment by hendersonreed** [comment]
chiming in: I've been seeing crashes for the past week at least, primarily after tapping on push notifications or typing out messages. `Google Play system update` is from August 1.

**Comment by RodEsp** [comment]
Same as above, been having random crashes for a few weeks on a Pixel 7 Pro.
`Google Play system update` is August 1 too .

Here's a crash log I got, from two crashes that happened last night, using `adb logcat --buffer=crash`.
```
09-19 21:27:40.322 25796 25938 F libc    :
09-19 21:27:40.322 25796 25938 F libc    : stack corruption detected (-fstack-protector)
09-19 21:27:40.334 25796 25796 F libc    : stack corruption detected (-fstack-protector)
09-19 21:27:40.803 25796 25796 F libc    : Fatal signal 6 (SIGABRT), code -1 (SI_QUEUE) in tid 25796 (com.zulipmobile), pid 25796 (com.zulipmobile)
09-19 21:27:41.883 26127 26127 F DEBUG   : *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***
09-19 21:27:41.883 26127 26127 F DEBUG   : Build fingerprint: 'google/cheetah/cheetah:13/TQ3A.230805.001/10316531:user/release-keys'
09-19 21:27:41.883 26127 26127 F DEBUG   : Revision: 'MP1.0'
09-19 21:27:41.883 26127 26127 F DEBUG   : ABI: 'arm64'
09-19 21:27:41.883 26127 26127 F DEBUG   : Timestamp: 2023-09-19 21:27:41.015024983-0400
09-19 21:27:41.883 26127 26127 F DEBUG   : Process uptime: 16s
09-19 21:27:41.883 26127 26127 F DEBUG   : Cmdline: com.zulipmobile
09-19 21:27:41.883 26127 26127 F DEBUG   : pid: 25796, tid: 25796, name: com.zulipmobile  >>> com.zulipmobile <<<
09-19 21:27:41.883 26127 26127 F DEBUG   : uid: 10440
09-19 21:27:41.883 26127 26127 F DEBUG   : tagged_addr_ctrl: 0000000000000001 (PR_TAGGED_ADDR_ENABLE)
09-19 21:27:41.883 26127 26127 F DEBUG   : signal 6 (SIGABRT), code -1 (SI_QUEUE), fault addr --------
09-19 21:27:41.883 26127 26127 F DEBUG   :     x0  0000000000000000  x1  00000000000064c4  x2  0000000000000006  x3  b400007d500da680
09-19 21:27:41.883 26127 26127 F DEBUG   :     x4  607372652c271f63  x5  607372652c271f63  x6  607372652c271f63  x7  7f7f7f7f7f7f7f7f
09-19 21:27:41.883 26127 26127 F DEBUG   :     x8  00000000000000f0  x9  0000007deb9179e0  x10 0000000000000001  x11 0000007deb959370
09-19 21:27:41.883 26127 26127 F DEBUG   :     x12 0000007e0626e020  x13 000000007fffffff  x14 0000000008c22b82  x15 000007e8d97b033c
09-19 21:27:41.883 26127 26127 F DEBUG   :     x16 0000007deb9c6d50  x17 0000007deb9a1eb0  x18 0000007e05afc000  x19 00000000000064c4
09-19 21:27:41.883 26127 26127 F DEBUG   :     x20 00000000000064c4  x21 00000000ffffffff  x22 0000007e04f9e000  x23 b400007d400f3090
09-19 21:27:41.883 26127 26127 F DEBUG   :     x24 b400007d400f3090  x25 0000007e04f9e000  x26 0000007ad25d0f20  x27 000000000000000b
09-19 21:27:41.883 26127 26127 F DEBUG   :     x28 0000000008000004  x29 b400007d500da700
09-19 21:27:41.883 26127 26127 F DEBUG   :     lr  0000007deb9491c8  sp  b400007d500da660  pc  0000007deb9491f4  pst 0000000000001000
09-19 21:27:41.883 26127 26127 F DEBUG   : backtrace:
09-19 21:27:41.883 26127 26127 F DEBUG   :       #00 pc 00000000000531f4  /apex/com.android.runtime/lib64/bionic/libc.so (abort+164) (BuildId: dc4001c2ef2dfc23467040797a96840c)
09-19 21:27:41.883 26127 26127 F DEBUG   :       #01 pc 0002b1970002b192  <unknown>
09-20 00:59:51.237 26157 26206 F libc    : stack corruption detected (-fstack-protector)
09-20 00:59:51.238 26157 26206 F libc    : stack corruption detected (-fstack-protector)
09-20 00:59:51.425 26157 26206 F libc    : Fatal signal 6 (SIGABRT), code -1 (SI_QUEUE) in tid 26206 (mqt_native_modu), pid 26157 (com.zulipmobile)
09-20 00:59:52.014 22770 22770 F DEBUG   : *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***
09-20 00:59:52.015 22770 22770 F DEBUG   : Build fingerprint: 'google/cheetah/cheetah:13/TQ3A.230805.001/10316531:user/release-keys'
09-20 00:59:52.015 22770 22770 F DEBUG   : Revision: 'MP1.0'
09-20 00:59:52.015 22770 22770 F DEBUG   : ABI: 'arm64'
09-20 00:59:52.015 22770 22770 F DEBUG   : Timestamp: 2023-09-20 00:59:51.654086572-0400
09-20 00:59:52.015 22770 22770 F DEBUG   : Process uptime: 12728s
09-20 00:59:52.015 22770 22770 F DEBUG   : Cmdline: com.zulipmobile
09-20 00:59:52.015 22770 22770 F DEBUG   : pid: 26157, tid: 26206, name: mqt_native_modu  >>> com.zulipmobile <<<
09-20 00:59:52.015 22770 22770 F DEBUG   : uid: 10440
09-20 00:59:52.015 22770 22770 F DEBUG   : tagged_addr_ctrl: 0000000000000001 (PR_TAGGED_ADDR_ENABLE)
09-20 00:59:52.015 22770 22770 F DEBUG   : signal 6 (SIGABRT), code -1 (SI_QUEUE), fault addr --------
09-20 00:59:52.015 22770 22770 F DEBUG   : Abort message: 'pdata'
09-20 00:59:52.015 22770 22770 F DEBUG   :     x0  0000000000000000  x1  000000000000665e  x2  0000000000000006  x3  b400007d500ca5e0
09-20 00:59:52.015 22770 22770 F DEBUG   :     x4  607372652c271f63  x5  607372652c271f63  x6  607372652c271f63  x7  7f7f7f7f7f7f7f7f
09-20 00:59:52.015 22770 22770 F DEBUG   :     x8  00000000000000f0  x9  0000007deb9179e0  x10 0000000000000001  x11 0000007deb959370
09-20 00:59:52.015 22770 22770 F DEBUG   :     x12 0000007e0626e020  x13 000000007fffffff  x14 00000000090147fe  x15 00000831b244021d
09-20 00:59:52.015 22770 22770 F DEBUG   :     x16 0000007deb9c6d50  x17 0000007deb9a1eb0  x18 0000007abd504000  x19 000000000000662d
09-20 00:59:52.015 22770 22770 F DEBUG   :     x20 000000000000665e  x21 00000000ffffffff  x22 0000000000000400  x23 0000007abd917000
09-20 00:59:52.015 22770 22770 F DEBUG   :     x24 b400007d500ca6f8  x25 0000007e04f9e000  x26 0000007dff28f358  x27 0000007dff2909b0
09-20 00:59:52.015 22770 22770 F DEBUG   :     x28 0000007dff28f748  x29 b400007d500ca660
09-20 00:59:52.015 22770 22770 F DEBUG   :     lr  0000007deb9491c8  sp  b400007d500ca5c0  pc  0000007deb9491f4  pst 0000000000001000
09-20 00:59:52.015 22770 22770 F DEBUG   : backtrace:
09-20 00:59:52.015 22770 22770 F DEBUG   :       #00 pc 00000000000531f4  /apex/com.android.runtime/lib64/bionic/libc.so (abort+164) (BuildId: dc4001c2ef2dfc23467040797a96840c)
09-20 00:59:52.015 22770 22770 F DEBUG   :       #01 pc 000000000006a3e4  /apex/com.android.runtime/lib64/bionic/libc.so (__stack_chk_fail+20) (BuildId: dc4001c2ef2dfc23467040797a96840c)
09-20 00:59:52.015 22770 22770 F DEBUG   :       #02 pc 000000000004e264  /apex/com.android.runtime/lib64/bionic/libc.so (async_safe_fatal_va_list+436) (BuildId: dc4001c2ef2dfc23467040797a96840c)
09-20 00:59:52.015 22770 22770 F DEBUG   :       #03 pc b400007d500cae30  <unknown>
```

**Comment by quinn-dougherty** [comment]
Galaxy S23, new. Unusably bad crashing. 

Android 13
Google Play System Update Aug 1

**Comment by gnprice** [comment]
Thanks everyone for the reports!

We've investigated this intensively over the past few days, and we've just released a version of Zulip that fixes these crashes: v27.213. If you're [in the beta](https://github.com/zulip/zulip-mobile#using-the-beta), the update is available now; if not, it'll be available once Play Store review is complete, which typically takes under a day.

For more details, read on.

---

This turns out to be an issue that's appearing in a number of different apps by different developers. It's being investigated in the issue trackers for Sentry, which is an error-reporting service we use, and for Android upstream:
* https://github.com/getsentry/sentry-unity/issues/1444
* https://issuetracker.google.com/issues/300840851

The conditions that trigger the bug appear to be:
* The device has taken the Google Play system update dated August 2023.
  * This is the part that's new — it seems like this update has only started rolling out widely in the past week.
  * There's also a September 2023 version, but it appears to trigger the bug the same as the August version.
* The device is running Android 12, 13, or 14; not Android 11 or older.
* The app is using Sentry.

It's not yet clear whether the bug is in Sentry or in the Android system components that are updated in the Google Play system update. The issue persists even with the latest versions of the Sentry libraries, as we determined at #5765.

In order to make these crashes stop happening for everyone using Zulip, our v27.213 release simply disables Sentry on Android. This isn't a super comfortable state to be in — error reporting is important for us to learn of other issues and fix them — but it lets us resolve the crashes while the Sentry folks work on a diagnosis and fix. I'll file a follow-up issue for re-enabling Sentry without the crashes.
